### ISSUE TITLE
Correlated sampling issue

### ISSUE THREAD

**Issue by pakoliver** [nrc-cnrc/EGSnrc#10]
When calculating doses to spherical cavities with radii on the order of microns, the use of correlated sampling causes the dose to be 2-4% smaller than expected. This discrepancy was observed for monoenergetic photon beams with energies on the order of tens of keV, incident on a cubic phantom with a side length of 2 cm. The amount of discrepancy depends on the energy and medium used. This does not seem to be a problem for larger scoring regions (radii > 10 microns).

**Comment by ftessier** [comment]
What is the size of your source? @MartinMartinov recently found that the default single-precision resolution for the random number generator in the particle sources is not sufficient, thus at the micron scale you end up sampling source particles on a discrete grid.

**Comment by pakoliver** [comment]
My source is parallel beam of monoenergetic photons with a radius of 1 cm. I am using the “high resolution = yes” option, as suggested by Martin Martinov.


On Nov 23, 2015, at 4:10 PM, Frédéric Tessier <notifications@github.com<mailto:notifications@github.com>> wrote:


What is the size of your source? @MartinMartinov<https://github.com/MartinMartinov> recently found that the default single-precision resolution for the random number generator in the particle sources is not sufficient, thus at the micron scale you end up sampling source particles on a discrete grid.

**Comment by ftessier** [comment]
@pakoliver, can you send me a minimal sample input file which illustrates the problem?

**Comment by pakoliver** [comment]
Hi Frederic,

I just wanted to give you a quick update so that you know I’ve seen your message:

So far, I’ve only seen the issue in my most complicated input files, which might be hard to follow for anyone but myself. Over the weekend, I ran a test with correlated sampling using a very simple geometry, and I was not able to reproduce the issue. I’d like to run a couple more tests with slightly more complicated simulation geometries to see if I can reproduce the issue there. These tests might take a couple weeks to do, depending on cluster availability.

Patty


On Jan 8, 2016, at 4:13 PM, Frédéric Tessier <notifications@github.com<mailto:notifications@github.com>> wrote:


@pakoliver<https://github.com/pakoliver>, can you send me a minimal sample input file which illustrates the problem?

**Comment by ftessier** [comment]
Thank you @pakoliver. Indeed if you can provide a simplified input that suffers from this problem it would be extremely useful. Perhaps you can simplify your complicated case step-by-step and note when the issue you raise vanishes; that in itself might provide the best clue as to where lies the problem.
